evennia/utils/tests/test_containers.py

```diff
@@ -63,19 +92,45 @@ def test_start_with_typeclassed_script_loads_it(self):
     self.assertEqual(gsc.typeclass_storage['script_name'], GoodScript)
 
   @override_settings(GLOBAL_SCRIPTS={'script_name': {'typeclass': 'evennia.utils.tests.test_containers.BadScript'}})
-  def test_start_with_bad_typeclassed_script_skips_it(self):
+  def test_start_with_bad_typeclassed_script_raises(self):
+    gsc = containers.GlobalScriptContainer()
+
+    with self.assertRaises(Exception) as cm:
+        gsc.start()
+        self.fail("An exception was expected but it didn't occur.")
+
+    self.assertEqual(len(gsc.typeclass_storage), 1)
+    self.assertIn('script_name', gsc.typeclass_storage)
+
+  @override_settings(GLOBAL_SCRIPTS={'script_name': {'typeclass': 'evennia.utils.tests.test_containers.QuestionableScript'}})
+  def test_start_with_questionably_subclassed_script_skips_it(self):
+    # Make sure _BASE_SCRIPT_TYPECLASS is NOT DefaultScript but it's a subclass
+    assert issubclass(_BASE_SCRIPT_TYPECLASS, DefaultScript) and _BASE_SCRIPT_TYPECLASS != DefaultScript
+
     gsc = containers.GlobalScriptContainer()
 
     gsc.start()
 
-    self.assertEqual(len(gsc.typeclass_storage), 0)
-    self.assertNotIn('script_name', gsc.typeclass_storage)
+    self.assertEqual(len(gsc.typeclass_storage), 1)
+    self.assertIn('script_name', gsc.typeclass_storage)
+    self.assertEqual(gsc.typeclass_storage['script_name'], QuestionableScript)
+
+  @override_settings(GLOBAL_SCRIPTS={'script_name': {'typeclass': 'evennia.utils.tests.test_containers.WorseScript'}})
+  def test_start_with_worse_typeclassed_script_raises(self):
+    gsc = containers.GlobalScriptContainer()
+
+    with self.assertRaises(Exception) as cm:
+        gsc.start()
+        self.fail("An exception was expected but it didn't occur.")
+
+    self.assertEqual(len(gsc.typeclass_storage), 1)
+    self.assertIn("script_name", gsc.typeclass_storage)
+    self.assertEqual(gsc.typeclass_storage["script_name"], WorseScript)
 
-  @override_settings(GLOBAL_SCRIPTS={'script_name': {'typeclass': 'evennia.utils.tests.test_containers.WorstScript'}})
-  def test_start_with_worst_typeclassed_script_skips_it(self):
+  @override_settings(GLOBAL_SCRIPTS={'script_name': {'typeclass': 'evennia.utils.tests.test_containers.NonexistentScript'}})
+  def test_start_with_nonexistent_script_skips_it(self):
     gsc = containers.GlobalScriptContainer()
 
     gsc.start()
 
     self.assertEqual(len(gsc.typeclass_storage), 0)
```

Comment:
`NonexistentScript` is  "loadable" but `start()` did not fail? Another quirk with a different outcome when compared with `BadScript` or `WorseScript`, IMHO.